### PR TITLE
Add test for exception throw inside Reverse PInvoke

### PR DESCRIPTION
**Pull request by kant2002**
This test works on CoreCLR but fails at NativeAOT currently

### CODE REVIEW

File: src/tests/Exceptions/ForeignThread/ForeignThreadExceptions.cs [hunk 1]
```diff
@@ -32,6 +32,17 @@ public static void RunTest()
...
 
+        try
+        {
+            InvokeCallback(() => {
+                MethodThatThrows();
+            });
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine("Caught exception thrown inside unmanaged code in a function called by a delegate called through Reverse PInvoke.");
```
Comment (jkotas) [inline src/tests/Exceptions/ForeignThread/ForeignThreadExceptions.cs:43]: This is testing Windows-specific behavior. This test will fail on Unix on all runtimes.

**Comment by jkotas** [comment]
> This test works on CoreCLR but fails at NativeAOT currently

Yes, that's expected. The Windows-specific exception handling interop is not supported in NativeAOT.

File: src/tests/Exceptions/ForeignThread/ForeignThreadExceptions.cs (hunk 1 above)
Comment (kant2002) [reply src/tests/Exceptions/ForeignThread/ForeignThreadExceptions.cs:43]: I'm puzzled. CI green, locally on Unix test fails. I check `issues.targets` and this test seems to be should run on CI. This gives me 2 questions
- why CI green where it should not
- What's technical details which make that scenario on Unix unreasonable choice?

Comment (jkotas) [reply src/tests/Exceptions/ForeignThread/ForeignThreadExceptions.cs:43]: > why CI green where it should not

I do not see green CI. I see a lot of yellow that means the CI is still running. I expect the yellow to turn into red once it finishes.

> What's technical details which make that scenario on Unix unreasonable choice?

The short description is at: https://docs.microsoft.com/en-us/dotnet/standard/native-interop/exceptions-interoperability

The long description is at: https://www.mono-project.com/docs/advanced/pinvoke/#runtime-exception-propagation

Comment (kant2002) [reply src/tests/Exceptions/ForeignThread/ForeignThreadExceptions.cs:43]: Is there some value in ejecting this code in a separate folder and make test Windows only? Or better close this altogether?

Comment (jkotas) [reply src/tests/Exceptions/ForeignThread/ForeignThreadExceptions.cs:43]: I think we have number of tests for validating the Windows-specific exception handling interop. For example here: https://github.com/dotnet/runtime/blob/ec0c7f3a91b9fc47f7a867f0b997ba34560302e7/src/tests/Interop/UnmanagedCallersOnly/UnmanagedCallersOnlyTest.cs#L57

**Comment by kant2002** [comment]
Thank you for showing me the ropes!

**Pull request closed without merging**